core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -364,15 +423,15 @@ class BrokerToControllerRequestThread(
   }
 
   override def doWork(): Unit = {
-    if (activeControllerAddress().isDefined) {
+    if (activeControllerOpt().isDefined) {
       super.pollOnce(Long.MaxValue)
     } else {
       debug("Controller isn't cached, looking for local metadata changes")
       controllerNodeProvider.get() match {
-        case Some(controllerNode) =>
-          info(s"Recorded new controller, from now on will use broker $controllerNode")
-          updateControllerAddress(controllerNode)
-          metadataUpdater.setNodes(Seq(controllerNode).asJava)
+        case Some(controllerNodeAndEpoch) =>
```

Comment:
Is this where/how eventually the `LeaderAndIsr` from the new controller change gets applied?